ios/Classes/SwiftCallKeepPlugin.swift

```diff
@@ -1,14 +1,521 @@
+    static let ACTION_CALL_TOGGLE_DMTF = "co.doneservices.callkeep.ACTION_CALL_TOGGLE_DMTF"
+    static let ACTION_CALL_TOGGLE_GROUP = "co.doneservices.callkeep.ACTION_CALL_TOGGLE_GROUP"
+    static let ACTION_CALL_TOGGLE_AUDIO_SESSION = "co.doneservices.callkeep.ACTION_CALL_TOGGLE_AUDIO_SESSION"
+    
+    @objc public static var sharedInstance: SwiftCallKeepPlugin? = nil
+    
+    private var channel: FlutterMethodChannel? = nil
+    private var eventChannel: FlutterEventChannel? = nil
+    private var callManager: CallManager? = nil
+    
+    private var eventCallbackHandler: EventCallbackHandler?
+    private var sharedProvider: CXProvider? = nil
+    
+    private var outgoingCall : Call?
+    private var answerCall : Call?
+    
+    private var data: Data?
+    private var isFromPushKit: Bool = false
+    private let devicePushTokenVoIP = "DevicePushTokenVoIP"
+    
+    private func sendEvent(_ event: String, _ body: [String : Any?]?) {
+        let data = body ?? [:] as [String : Any?]
+        eventCallbackHandler?.send(event, data)
+    }
+    
+    @objc public func sendEventCustom(_ event: String, body: Any?) {
+        eventCallbackHandler?.send(event, body ?? [:] as [String : Any?])
+    }
+    
+    public static func sharePluginWithRegister(with registrar: FlutterPluginRegistrar) -> SwiftCallKeepPlugin {
+        if(sharedInstance == nil){
+            sharedInstance = SwiftCallKeepPlugin()
+        }
+        sharedInstance!.channel = FlutterMethodChannel(name: "flutter_callkeep", binaryMessenger: registrar.messenger())
+        sharedInstance!.eventChannel = FlutterEventChannel(name: "flutter_callkeep_events", binaryMessenger: registrar.messenger())
+        sharedInstance!.callManager = CallManager()
+        sharedInstance!.eventCallbackHandler = EventCallbackHandler()
+        sharedInstance!.eventChannel?.setStreamHandler(sharedInstance!.eventCallbackHandler as? FlutterStreamHandler & NSObjectProtocol)
+        return sharedInstance!
+    }
+    
+    
+    public static func register(with registrar: FlutterPluginRegistrar) {
+        let instance = sharePluginWithRegister(with: registrar)
+        registrar.addMethodCallDelegate(instance, channel: instance.channel!)
+    }
+    
+    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        switch call.method {
+        case "displayIncomingCall":
+            guard let args = call.arguments else {
+                result("OK")
+                return
+            }
+            if let getArgs = args as? [String: Any] {
+                self.data = Data(args: getArgs)
+                displayIncomingCall(self.data!, fromPushKit: false)
+            }
+            result("OK")
+            break
+        case "showMissCallNotification":
+            result("OK")
+            break
+        case "startCall":
+            guard let args = call.arguments else {
+                result("OK")
+                return
+            }
```

Comment:
This is weird. In all of these, the plugin returns "OK" if it can't continue. Shouldn't it at least return "Missing arguments" or something?

Reply:
True, it should only return `OK` for Androind only methods that it shouldn't handle, otherwise it should throw something meaningful back.